library/ssl_cli.c

```diff
@@ -687,7 +687,7 @@ static int ssl_write_session_ticket_ext( mbedtls_ssl_context *ssl,
         return( 0 );
 
     MBEDTLS_SSL_DEBUG_MSG( 3,
-        ( "sending session ticket of length %d", tlen ) );
+        ( "sending session ticket of length %zu", tlen ) );
```

Comment:
Is there a GCC or Clang option (`-Wformat=…`) that we could enable to avoid such errors in the future?

Reply:
Yes, as it turns out we were not actually getting ANY warnings for MBEDTLS_SSL_DEBUG_MSG. The scope of this patch has just changed slightly (found a tonne more warnings), so will push second pass when that is done